todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/statistics/StatisticsViewModel.java

```diff
@@ -16,72 +16,55 @@
 
 package com.example.android.architecture.blueprints.todoapp.statistics;
 
+
 import android.support.annotation.NonNull;
 import android.support.v4.util.Pair;
 
 import com.example.android.architecture.blueprints.todoapp.data.Task;
 import com.example.android.architecture.blueprints.todoapp.data.source.TasksRepository;
 import com.example.android.architecture.blueprints.todoapp.util.EspressoIdlingResource;
-import com.example.android.architecture.blueprints.todoapp.util.schedulers.BaseSchedulerProvider;
 
 import java.util.List;
 
 import rx.Observable;
-import rx.Subscription;
 import rx.functions.Action0;
-import rx.functions.Action1;
 import rx.functions.Func1;
 import rx.functions.Func2;
-import rx.subscriptions.CompositeSubscription;
+import rx.subjects.PublishSubject;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
 /**
  * Listens to user actions from the UI ({@link StatisticsFragment}), retrieves the data and updates
  * the UI as required.
  */
-public class StatisticsPresenter implements StatisticsContract.Presenter {
+public class StatisticsViewModel {
 
     @NonNull
     private final TasksRepository mTasksRepository;
 
     @NonNull
-    private final StatisticsContract.View mStatisticsView;
-
-    @NonNull
-    private final BaseSchedulerProvider mSchedulerProvider;
-
-    @NonNull
-    private CompositeSubscription mSubscriptions;
+    private PublishSubject<Boolean> mProgressIndicatorSubject;
```

Comment:
Shouldn't this be a ``BehaviorSubject`` with default ``false``. So every time you subscribe to it, you know the exact state of the progress